Automattic/wp-calypso
# Store: Add the ability to update a review's status and delete a review.

This PR adds the code necessary to manage a review's status, including permanently deleting it from the 'trash' queue. 

This is based on the design from #14177 with some differences to bring the experience inline with the new comments interface, which this PR is patterned off of.

To Test:
* Make sure you are the latest master of `wc-api-dev` (API changes have been merged). (I'll be getting this deployed for AT sites shortly).
* Run `npm run test-client client/extensions/woocommerce/state` and make sure all tests pass.
* Make sure you have a few reviews, including at least 1 pending review.
* Go to `http://calypso.localhost:3000/store/reviews/:site`.
* Verify that you can see your pending review. Expand it to open up the action bar, and select "Approve".
* Verify that your review approves. You should see a success message and the entry should change colors.
* Open the review again and trash it. You should again see a success message, and it should leave the queue and go into the trash tab.
* Go to the 'Trash' queue, and find the review. Click 'Delete' to actually delete the review.

Depends on a fix from #18168.
